src/smolagents/agents.py

```diff
@@ -1019,6 +1019,44 @@ def step(self, memory_step: ActionStep) -> Union[None, Any]:
             level=LogLevel.DEBUG,
         )
 
+        # Re-try if the model did not call any tools
+        if (model_message.content != '') and (model_message.tool_calls is None or len(model_message.tool_calls) == 0):
```

Comment:
You can write:

```py
if (model_message.content.strip() != '') and not model_message.tool_calls:
    ...
```

The `not model_message.tool_calls` covers both `None` and empty list cases.

You should `strip` `content` to cover the case that `content` has some data but only whitespaces.